c10/core/DispatchKeySet.h

```diff
@@ -168,27 +219,30 @@ class DispatchKeySet final {
     DispatchKey operator*() const { return static_cast<DispatchKey> (i_); }
 
    private:
-    const uint64_t *data_ptr_;
+    const DispatchKeySet *data_ptr_;
     uint8_t i_;
   };
 
  public:
   // Returns iterator to the first key in the set. If no keys are in the
   // set, then will return the end iterator.
-  iterator begin() const { return iterator(&repr_); }
+  iterator begin() const { return iterator(&ks_, has_undefined_); }
 
   // We do not need to iterate beyond NumDispatchKeys so we will treat this as
   // the end iterator. NumDispatchKeys will always be strictly less than 64.
-  iterator end() const { return iterator(&repr_, static_cast<uint8_t>(DispatchKey::NumDispatchKeys)); }
+  iterator end() const { return iterator(&ks_, has_undefined_, static_cast<uint8_t>(DispatchKey::NumDispatchKeys)); }
 
+private:
+  DispatchKeySet ks_;
+  bool has_undefined_;
 };
 
 C10_API std::string toString(DispatchKeySet);
 C10_API std::ostream& operator<<(std::ostream&, DispatchKeySet);
 
 // autograd_dispatch_keyset should include all runtime autograd keys.
 // Alias key DispatchKey::Autograd maps to autograd_dispatch_keyset.
-constexpr DispatchKeySet autograd_dispatch_keyset = DispatchKeySet({
+constexpr RuntimeDispatchKeySet autograd_dispatch_keyset ({
```

Comment:
Mmmmm... why make this a RuntimeDispatchKeySet? It's a valid DispatchKeySet and it's better to have a more precise type. (I got here because I noticed a runtime `key_set()` call where there wasn't one before.)